### PR TITLE
hibernation option to compute

### PR DESCRIPTION
**Pull request by taniahagan**
Add the ability to enable hibernation on instances and activate hibernation

### CODE REVIEW

File: lib/fog/aws/requests/compute/stop_instances.rb [hunk 1]
```diff
@@ -16,9 +16,13 @@ class Real
         #     * TODO: fill in the blanks
         #
         # {Amazon API Reference}[http://docs.amazonwebservices.com/AWSEC2/latest/APIReference/ApiReference-query-StopInstances.html]
-        def stop_instances(instance_id, force = false)
+        def stop_instances(instance_id, options = {})
           params = Fog::AWS.indexed_param('InstanceId', instance_id)
-          params.merge!('Force' => 'true') if force
+          params.merge!('Force' => 'true') if options['force']
```
Comment (geemus) [inline lib/fog/aws/requests/compute/stop_instances.rb:21]: For backwards compatibility, could you tweak this to check to see if options is a boolean value, and if so print a warning and then automatically convert it to the appropriate hash? I think that should strike a good balance of adding the new functionality while also preventing a breaking change for existing users. Does that make sense?

Comment (taniahagan) [reply lib/fog/aws/requests/compute/stop_instances.rb:21]: It does, I shall take a look at it

Comment (taniahagan) [reply lib/fog/aws/requests/compute/stop_instances.rb:25]: Ok, added the change

Comment (geemus) [reply lib/fog/aws/requests/compute/stop_instances.rb:25]: Looks great, thanks!

File: lib/fog/aws/requests/compute/run_instances.rb [hunk 2]
```diff
@@ -221,6 +232,7 @@ def run_instances(image_id, min_count, max_count, options = {})
               'associatePublicIP'     => options['associatePublicIP'] || false,
               'architecture'          => 'i386',
               'blockDeviceMapping'    => block_device_mapping,
+              'hibernationOptions'    => hibernation_options,
```
Comment (geemus) [inline lib/fog/aws/requests/compute/run_instances.rb:235]: Hey, I'm not sure why the integration isn't working right, but travis-ci is reporting an error here: https://travis-ci.org/github/fog/fog-aws/jobs/699781152#L413. In particular, `hibernation_options` is referred to is this mock version of the method without having previously been defined. We may need to either refer directly to the options hash or do something similar to the delete above. Does that make sense? Thanks!

Comment (taniahagan) [reply lib/fog/aws/requests/compute/run_instances.rb:243]: Hopefully this change should do it.  Tested successfully running local mock test.

**Comment by geemus** [comment]
Thanks for working through that with me. Do you need a new release with these changes?

**Comment by taniahagan** [comment]
A new release would be good, though if one is due soon happy to wait a while longer.

**Comment by geemus** [comment]
Released in v3.6.6. Thanks!

**Comment by taniahagan** [comment]
Excellent, thank you very much.

**Pull request merged**